lib/backend/k8s/k8s.go

```diff
@@ -214,6 +219,29 @@ func (c *KubeClient) getProfile(k model.ProfileKey) (*model.KVPair, error) {
 	return c.converter.namespaceToProfile(namespace)
 }
 
+// applyWorkloadEndpoint patches the existing Pod to include an IP address, if
+// one has been set on the workload endpoint.
+func (c *KubeClient) applyWorkloadEndpoint(k *model.KVPair) (*model.KVPair, error) {
+	ips := k.Value.(*model.WorkloadEndpoint).IPv4Nets
+	if len(ips) > 0 {
+		log.Debugf("Applying workload with IPs: %+v", ips)
+		ns, name := c.converter.parseWorkloadID(k.Key.(model.WorkloadEndpointKey).WorkloadID)
+		pod, err := c.clientSet.Pods(ns).Get(name)
+		if err != nil {
+			return nil, err
+		}
+		pod.Status.PodIP = ips[0].IP.String()
+		log.Debugf("Pod IP: %+v", pod.Status.PodIP)
+		pod, err = c.clientSet.Pods(ns).UpdateStatus(pod)
```

Comment:
I think we should be converting the error here to a known error type for the Calico API.   In particular it feels that we should explicitly handle a resource update conflict in the CNI plugin (so that if k8s sorts itself out and magically updates before we do then we could hit concurrent update issues).